### PR TITLE
First pass at collections outline

### PR DESCRIPTION
**Pull request by tmeasday**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465272%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465312%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465337%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465459%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465480%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465489%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465508%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465517%22%2C%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465572%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20f24b32dc30ab1697ec00a44f09c7e041f5ee495e%20outlines/collections.md%2039%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465312%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20don%27t%20know%20the%20answer%20to%20this%20question%2C%20but%20we%20should%20figure%20it%20out.%20Perhaps%20we%20can%20leave%20a%20question%20mark%20for%20the%20outline%20for%20now.%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A51%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yeah.%20%40zhenyasav%20or%20%40zol%20may%20have%20some%20up-to-date%20thoughts%20on%20this%3F%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A55%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/collections.md%3AL1-55%22%7D%2C%20%22Pull%20f24b32dc30ab1697ec00a44f09c7e041f5ee495e%20outlines/collections.md%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465337%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20this%20just%20docs%20for%20SimpleSchema%3F%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A52%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22More%20or%20less.%20Should%20we%20skip%20it%3F%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A56%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22Probably%20-%20if%20there%20is%20a%20good%20place%20to%20link%20to%20for%20this%20content.%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A56%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A58%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/collections.md%3AL1-55%22%7D%2C%20%22Pull%20f24b32dc30ab1697ec00a44f09c7e041f5ee495e%20outlines/collections.md%2027%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/61%23discussion_r43465272%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22No%20doubt%20here%21%20Always%20do%20%60addFoo%60%2C%20never%20use%20%60update%60%2C%20_ever_%2C%20IMO.%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A50%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%2C%20will%20update%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A55%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-10-30T01%3A56%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/collections.md%3AL1-55%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull f24b32dc30ab1697ec00a44f09c7e041f5ee495e outlines/collections.md 39'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/61#discussion_r43465312'>File: outlines/collections.md:L1-55</a></b>
- <a href='https://github.com/stubailo'></a> I don't know the answer to this question, but we should figure it out. Perhaps we can leave a question mark for the outline for now.
- <a href='https://github.com/tmeasday'></a> Yeah. @zhenyasav or @zol may have some up-to-date thoughts on this?
- [x] <a href='#crh-comment-Pull f24b32dc30ab1697ec00a44f09c7e041f5ee495e outlines/collections.md 27'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/61#discussion_r43465272'>File: outlines/collections.md:L1-55</a></b>
- <a href='https://github.com/stubailo'></a> No doubt here! Always do `addFoo`, never use `update`, _ever_, IMO.
- <a href='https://github.com/tmeasday'></a> Ok, will update
- [x] <a href='#crh-comment-Pull f24b32dc30ab1697ec00a44f09c7e041f5ee495e outlines/collections.md 45'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/61#discussion_r43465337'>File: outlines/collections.md:L1-55</a></b>
- <a href='https://github.com/stubailo'></a> Is this just docs for SimpleSchema?
- <a href='https://github.com/tmeasday'></a> More or less. Should we skip it?
- <a href='https://github.com/stubailo'></a> Probably - if there is a good place to link to for this content.

<a href='https://www.codereviewhub.com/meteor/guide/pull/61?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/meteor/guide/pull/61?approve=1'></a>&nbsp;<a href='https://github.com/meteor/guide/pull/61'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by stubailo** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/meteor/guide/pull/61'>Refresh</a>.<a href='#crh-update-none'></a>

File: outlines/collections.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  4. "Hooking" data by subclassing Collection2.
+    1. Description of the need for hooks
+    2. How the careful use of utilities can allow readable mutators that have hooks
+  5. EG: Denormalization patterns
+    1. Define your denormalizer in a different file
+    2. Hook the denormalizer in various `insert/update/remove` functions
+4. Custom mutators
+  1. In a public API it's best to be *less* general rather than *more* general (see security article)
+  2. Your methods are your public API.
+  3. So, if in doubt, write a `bar.addFoo` mutator rather than allowing `bar.update` to add `foo`.
```
Comment (stubailo) [inline outlines/collections.md]: No doubt here! Always do `addFoo`, never use `update`, _ever_, IMO.

File: outlines/collections.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+4. Designing your data schema
+  1. "Impure" mongo -- i.e. things that Meteor will force you to do that you might not have done otherwise
+    - Avoid subdocuments and large changing properties
+    - Use more collections, normalize more
+  2. Thinking ahead to future database changes
+    - Don't try to predict the future but be flexible
+5. Changing data schema - how to use migrations
+  1. percolate:migrations package
+  2. How to run migrations against a production db
+    [is our best advice run locally pointing at the production db, use Meteor shell?]
```
Comment (stubailo) [inline outlines/collections.md]: I don't know the answer to this question, but we should figure it out. Perhaps we can leave a question mark for the outline for now.

File: outlines/collections.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  3. Multiple stage deploys which can handle both new and old format
+6. Relational data and other helpers
+  1. Using `dburles:collection-helpers` to add "methods" to your documents
+  2. Returning a cursors from a helper to get related documents
+  3. Using a `cursor-utils` package to narrow down cursors etc [HELP NEEDED?]
+7. Advanced schema usage
```
Comment (stubailo) [inline outlines/collections.md]: Is this just docs for SimpleSchema?

File: outlines/collections.md (hunk 1 above)
Comment (tmeasday) [inline outlines/collections.md]: Ok, will update

File: outlines/collections.md (hunk 2 above)
Comment (tmeasday) [inline outlines/collections.md]: Yeah. @zhenyasav or @zol may have some up-to-date thoughts on this?

File: outlines/collections.md (hunk 3 above)
Comment (tmeasday) [inline outlines/collections.md]: More or less. Should we skip it?

Comment (stubailo) [inline outlines/collections.md]: Probably - if there is a good place to link to for this content.

File: outlines/collections.md (hunk 1 above)
Comment (tmeasday) [inline outlines/collections.md]: :+1:

File: outlines/collections.md (hunk 3 above)
Comment (tmeasday) [inline outlines/collections.md]: :+1:

File: outlines/collections.md (hunk 2 above)
Comment (tmeasday) [inline outlines/collections.md]: Going to follow that up elsewhere and say we leave this for now.

**Pull request merged**